File: src/java/src/main/java/org/elastos/hive/vault/database/SortItem.java
package org.elastos.hive.vault.database;

import java.util.ArrayList;

abstract class SortItem extends ArrayList<Object> {
    protected enum Order {
        ASCENDING(1), DESCENDING(-1);

        private Integer value;

        Order(Integer value) {
            this.value = value;
        }

        public Integer getValue() {
            return value;
        }
    }

    protected SortItem(String key, Order order) {
        super.add(key);
        super.add(order.getValue());
    }
}

File: src/restclient/database/sortitem.ts

export class SortItem {
    public static ASCENDING = 1;
    public static DESCENDING = -1;

    public key: string;
    public order: number;

    public constuctor(key?: string, order?: number) {
        this.key = key;
        this.order = order;
    }
}
